File: rod.py
def rodCutting(length, prices):
    max_revenue = [0] * (length + 1)

    for i in range(1, length + 1):
        for j in range(i):
            max_revenue[i] = max(max_revenue[i], prices[j] + max_revenue[i - j - 1])

    return max_revenue[length]

length = int(input("Enter the length of the rod: "))
prices = list(map(int, input("Enter the prices separated by space: ").split()))

while len(prices) < length:
    print("Enter all prices equal to the length of the rod.")
    prices = list(map(int, input("Enter the prices again: ").split()))

max_revenue = rodCutting(length, prices)
print(f"Maximum profit is {max_revenue}")

File: rod.java
import java.util.Scanner;

public class rod {

    public static int rodCutting(int length, int[] prices) {
        int[] maxRevenue = new int[length + 1];

        for (int i = 1; i <= length; i++) {
            for (int j = 1; j <= i; j++) {
                maxRevenue[i] = Math.max(maxRevenue[i], prices[j - 1] + maxRevenue[i - j]);
            }
        }

        return maxRevenue[length];
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Taking input for rod length
        System.out.print("Enter the length of the rod: ");
        int length = scanner.nextInt();

        // Taking input for rod prices
        int[] prices = new int[length];
        System.out.print("Enter the prices separated by space: ");
        for (int i = 0; i < length; i++) {
            prices[i] = scanner.nextInt();
        }

        int maxRevenue = rodCutting(length, prices);
        System.out.println("Maximum revenue: " + maxRevenue);
    }
}
